### PR TITLE
VBE Braintree 2.4.0

### PR DESCRIPTION
**Pull request by AidanThreadgold**
## Purpose of this pull request

This pull request (PR) updates the Braintree Payments configuration documentation to include docs on the new/changed configuration options available in the bundled Braintree M2 VBE extension (BUNDLE-2561).

## Affected documentation pages

- https://docs.magento.com/user-guide/payment/braintree.html
- https://docs.magento.com/user-guide/configuration/sales/braintree.html
- https://docs.magento.com/user-guide/stores/compliance-payment-services-directive.html

whatsnew
The [Braintree](https://docs.magento.com/user-guide/payment/braintree.html) core integration is now replaced by the Braintree extension available on the Magento Marketplace. This extension is now included in the Magento release.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by leslietilling** [comment]
@AidanThreadgold  Please sign the Adobe CLA so that we can process this PR: http://opensource.adobe.com/cla.html

**Comment by AidanThreadgold** [comment]
> @AidanThreadgold Please sign the Adobe CLA so that we can process this PR: http://opensource.adobe.com/cla.html

@leslietilling I have signed & submitted the CLA
CC @magterskine

File: src/configuration/sales/braintree.md [hunk 1]
```diff
@@ -2,91 +2,209 @@
...
 |Sort Order|Website|Determines the order that Braintree is listed with other payment methods during checkout.|
 
+
+
+### Kount
+![]({% link images/images/config-sales-payment-methods-braintree-kount-configuration.png %}){: .zoom}
+_Kount Configuration_
+
+
+|Kount Configuration| | When Advanced Fraud Protection option will be enabled, Kount configuration option will appear|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:57]: Edit for what is there, but need to explain what this does. 

```suggestion
|Kount Configuration| | When the _Advanced Fraud Protection_ option is enabled, Kount configuration option appears.|
```

File: src/configuration/sales/braintree.md [hunk 2]
```diff
@@ -2,91 +2,209 @@
...
+|Bypass Fraud Protection Threshold|Website|Advanced fraud protection checks will be bypassed if this threshold is met or exceeded. Leaving this field blank will disable this option.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:58]:
```suggestion
|Bypass Fraud Protection Threshold|Website|Advanced fraud protection checks will be bypassed if this threshold is met or exceeded. Leaving this field blank disables this option.|
```

File: src/configuration/sales/braintree.md [hunk 3]
```diff
@@ -2,91 +2,209 @@
...
+|ENS URL|Website|Under Kount AWC control panel, ENS URL needs to be added for you|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:59]: I'm confused as to what this means. Do we mean that they need to enter the URL that is in the AWC control panel for their Kount account?

```suggestion
|ENS URL|Website|Under Kount AWC control panel, ENS URL needs to be added for you|
```

File: src/configuration/sales/braintree.md [hunk 4]
```diff
@@ -2,91 +2,209 @@
...
+|Enabled ACH Direct Debit|Website|Determines if PayPal is included as a payment method through Braintree. Options: Yes / No|
+
+## Apple Pay through Braintree
+
+![]({% link images/images/config-sales-payment-methods-braintree-applepay-payment-method-configuration.png %}){: .zoom}
+_Apple Pay through Braintree_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enabled ApplePay through Braintree|Website|Determines if ApplePay is included as a payment method through Braintree. Options: Yes / No <br/><br/> The Domain will need to be [verified in Braintree Account first] https://developers.braintreepayments.com/guides/apple-pay/configuration/javascript/v3.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:91]: link syntax needs a fix

```suggestion
|Enabled ApplePay through Braintree|Website|Determines if ApplePay is included as a payment method through Braintree. Options: Yes / No <br/><br/> The Domain will need to be [verified in Braintree Account first](https://developers.braintreepayments.com/guides/apple-pay/configuration/javascript/v3).|
```

File: src/configuration/sales/braintree.md [hunk 5]
```diff
@@ -2,91 +2,209 @@
...
+|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** (Previously "Authorize and Capture" before 2.4.0) - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:92]:
```suggestion
|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin. **_Note:_** This was  _Authorize and Capture_ in 2.3.x and earlier releases.|
```

File: src/configuration/sales/braintree.md [hunk 6]
```diff
@@ -2,91 +2,209 @@
...
+|Merchant Name|Store View|Label that will be displayed to customers in Apple Pay popup|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:93]:
```suggestion
|Merchant Name|Store View|Label that will be displayed to customers in the Apple Pay popup.|
```

File: src/configuration/sales/braintree.md [hunk 7]
```diff
@@ -2,91 +2,209 @@
...
+## Local Payment Methods
+
+![]({% link images/images/config-sales-payment-methods-braintree-local-payment-method-configuration.png %}){: .zoom}
+_Local Payment Methods_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enabled Local Payment Methods|Website|Determines if Local Payment Method is included as a payment method through Braintree. Options: Yes / No|
+|Title|Website|Label that will appear on the checkout Payment Method section. Default value: Local|
+|Allowed Payment Method|Website|Select the local Payment method which needs to be enabled.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:104]:
```suggestion
|Allowed Payment Method|Website|Select the local Payment method that needs to be enabled.|
```

File: src/configuration/sales/braintree.md (hunk 6 above)
Comment (leslietilling) [inline src/configuration/sales/braintree.md:93]:
```suggestion
|Merchant Name|Store View|Label that will be displayed to customers in the Apple Pay popup.|
```

File: src/configuration/sales/braintree.md [hunk 8]
```diff
@@ -2,91 +2,209 @@
...
+
+## GooglePay through Braintree
+
+![]({% link images/images/config-sales-payment-methods-braintree-googlepay-payment-method-configuration.png %}){: .zoom}
+_GooglePay through Braintree_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enabled GooglePay through Braintree|Website|Determines if Google Pay payment is included as a payment method through Braintree. Options: Yes / No|
+|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** (Previously "Authorize and Capture" before 2.4.0) - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:114]:
```suggestion
|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin. **_Note:_** This was  _Authorize and Capture_ in 2.3.x and earlier releases.|
```

File: src/configuration/sales/braintree.md [hunk 9]
```diff
@@ -2,91 +2,209 @@
...
+|Button Color|Website|Determines the color of Google Pay button. Option: White / Black|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:115]:
```suggestion
|Button Color|Website|Determines the color of the Google Pay button. Options: White / Black|
```

File: src/configuration/sales/braintree.md [hunk 10]
```diff
@@ -2,91 +2,209 @@
...
+|Merchant ID|Store View|ID provided by Google needs to be mentioned here|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:116]:
```suggestion
|Merchant ID|Store View|ID provided by Google must be entered here.|
```

File: src/configuration/sales/braintree.md [hunk 11]
```diff
@@ -2,91 +2,209 @@
...
+|Accepted Cards|Website|Select the type of cards which customer can use to place order using Google Pay|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:117]:
```suggestion
|Accepted Cards|Website|Select the type of cards that a customer can use to place order using Google Pay.|
```

File: src/configuration/sales/braintree.md [hunk 12]
```diff
@@ -2,91 +2,209 @@
...
+
+## Venmo through Braintree
+
+![]({% link images/images/config-sales-payment-methods-braintree-venmo-payment-method-configuration.png %}){: .zoom}
+_Venmo through Braintree_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enabled Venmo through Braintree|Website|Determines if Venmo is included as a payment method through Braintree. Options: Yes / No|
+|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** (Previously "Authorize and Capture" before 2.4.0) - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin.|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:128]:
```suggestion
|Payment Action|Website|Determines the action taken by Braintree when a payment is processed. Options: <br/>**Authorize** - Funds on the customer’s card are authorized, but not transferred from the customer’s account. An order is created in your store Admin. You can later capture the sale and create an invoice. <br/>**Intent Sale** - Funds on the customer’s card are authorized and captured by Braintree, and an order and invoice are created in your store Admin. **_Note:_** This was  _Authorize and Capture_ in 2.3.x and earlier releases.|
```

File: src/configuration/sales/braintree.md [hunk 13]
```diff
@@ -2,91 +2,209 @@
...
+
+### Mini-Cart and Cart Page
+
+![]({% link images/images/config-sales-payment-methods-braintree-Paypal-minicart-and-cart-configuration.png %}){: .zoom}
+_Mini cart and cart page_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Disable Funding Option|Website||
+|Customise Button|Website|If enabled, an option to customize the shape and color of the Paypal button will be available. Option: Yes / No|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:158]:
```suggestion
|Customise Button|Website|If enabled, an option to customize the shape and color of the Paypal button will be available. Options: Yes / No|
```

File: src/configuration/sales/braintree.md [hunk 14]
```diff
@@ -2,91 +2,209 @@
...
+|Shape|Website|Determine the shape of the Paypal button. Option: Pill / Rectangle|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:159]:
```suggestion
|Shape|Website|Determines the shape of the Paypal button. Options: Pill / Rectangle|
```

File: src/configuration/sales/braintree.md [hunk 15]
```diff
@@ -2,91 +2,209 @@
...
+|Size|Website|Determine the size of the Paypal button. Option: Medium / Large / Responsive|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:160]:
```suggestion
|Size|Website|Determines the size of the Paypal button. Options: Medium / Large / Responsive|
```

File: src/configuration/sales/braintree.md [hunk 16]
```diff
@@ -2,91 +2,209 @@
...
+|Color|Website|Determine the color of the Paypal button. Option: Blue / Black / Gold / Silver|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:161]:
```suggestion
|Color|Website|Determines the color of the Paypal button. Options: Blue / Black / Gold / Silver|
```

File: src/configuration/sales/braintree.md [hunk 17]
```diff
@@ -2,91 +2,209 @@
...
+
+
+### Checkout Page
+
+![]({% link images/images/config-sales-payment-methods-braintree-Paypal-checkout-configuration.png %}){: .zoom}
+_Checkout Page_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Disable Funding Option|Website||
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:171]: Description?

File: src/configuration/sales/braintree.md [hunk 18]
```diff
@@ -2,91 +2,209 @@
...
+|Display on Shopping Cart|Website|Determines if the PayPal button appears in the [mini cart]({% link sales/cart-sidebar.md %}) and on the [shopping cart]({% link sales/cart.md %}) page. Options: Yes / No|
+
+### Mini-Cart and Cart Page
+
+![]({% link images/images/config-sales-payment-methods-braintree-Paypal-minicart-and-cart-configuration.png %}){: .zoom}
+_Mini cart and cart page_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Disable Funding Option|Website||
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:157]: Description?

File: src/configuration/sales/braintree.md [hunk 19]
```diff
@@ -2,91 +2,209 @@
...
+
+### Checkout Page
+
+![]({% link images/images/config-sales-payment-methods-braintree-Paypal-checkout-configuration.png %}){: .zoom}
+_Checkout Page_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Disable Funding Option|Website||
+|Customise Button|Website|If enabled, an option to customize the shape and color of the Paypal button will be available. Option: Yes / No|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:172]: All these should be "Options:"

Comment (leslietilling) [inline src/configuration/sales/braintree.md:172]: All of these should be "Determines" and "Options:"

File: src/configuration/sales/braintree.md [hunk 20]
```diff
@@ -2,91 +2,209 @@
...
+
+
+### Product Page
+
+![]({% link images/images/config-sales-payment-methods-braintree-Paypal-product-page-configuration.png %}){: .zoom}
+_Product Page_
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Enable PayPal Buy Now button on the product page|Website|If enabled, the Paypal button will be available on the Product detail page. Option: Yes / No|
```
Comment (leslietilling) [inline src/configuration/sales/braintree.md:185]: All of these should be "Determines" (last three) and "Options:"

File: src/configuration/sales/braintree.md (hunk 19 above)
Comment (leslietilling) [inline src/configuration/sales/braintree.md:172]: All these should be "Options:"

File: src/configuration/sales/braintree.md [hunk 21]
```diff
@@ -2,91 +2,209 @@
...
-As of September 14, 2019, European banks might decline payments that do not meet [PSD2]({% link stores/compliance-payment-services-directive.md %}) requirements. Starting with Magento ver. 2.2.3, Braintree provides native support for 3D Secure 2.0.<br/>To meet PSD2 requirements for earlier versions of Magento, install and configure the official Braintree payment integration extension from [Magento Marketplace](https://marketplace.magento.com/catalogsearch/result/?q=braintree). Older Braintree implementations that run on JavaScript SDK v2 do not support 3D Secure 2.0.
+As of September 14, 2019, European banks might decline payments that do not meet [PSD2]({% link stores/compliance-payment-services-directive.md %}) requirements. Starting with Magento ver. 2.2.3, Braintree provides native support for 3D Secure 2.0.<br/>To meet PSD2 requirements for earlier versions of Magento, install and configure the official Braintree payment integration extension from [Magento Marketplace](https://marketplace.magento.com/catalogsearch/result/?q=braintree). Older Braintree implementations that run on JavaScript SDK v2 do not support 3D Secure 2.0.
```
[the remark this answers is not in the document]
Comment (mharron-adobe) [reply src/configuration/sales/braintree.md:9]: I agree @leslietilling  we can leave it out

File: src/configuration/sales/braintree.md [hunk 22]
```diff
@@ -2,91 +2,209 @@
...
+
+## 2.4 Migration
+Prior to Magento 2.4 it was recommended merchants install and configure the official Braintree payment integration extension from the [Magento Marketplace](https://marketplace.magento.com/catalogsearch/result/?q=braintree). As of Magento 2.4 the extension on the Magento Marketplace is now included in the core Magento release. <br/><br/>
+When migrating to Magento 2.4 merchants will need to uninstall the extension distributed on the Marketplace (`paypal/module-braintree` or `gene/module-braintree`) and update any code customisations to use the `PayPal_Braintree` namespace instead of `Magento_Braintree`. Configuration settings from the core Magento Braintree Payments extension and the extension distributed on the Magento Marketplace will be persisted and payments placed with those versions of the extension can still be captured, voided or refunded as normal.
```
[the remark this answers is not in the document]
Comment (mharron-adobe) [reply src/configuration/sales/braintree.md:13]: @leslietilling I agree with updates.

**Comment by leslietilling** [comment]
Thanks @AidanThreadgold  :). I can go ahead and apply edits if you are good with them and that is easier for you.

**Comment by keharper** [comment]
@AidanThreadgold Please review these topics on devdocs. If any changes are required, create a PR over there. If no changes are required, reply to this comment letting me know that:

https://devdocs.magento.com/guides/v2.4/graphql/payment-methods/braintree.html
https://devdocs.magento.com/guides/v2.4/graphql/payment-methods/braintree-vault.html

**Comment by AidanThreadgold** [comment]
Hey @leslietilling - I've applied the requested changes, please feel free to make any changes if that's easier for you :) 

@keharper No changes are needed to the GraphQL Devdoc links, thanks.

**Comment by leslietilling** [comment]
running tests

**Pull request merged**